src/presetsController.ts

```diff
@@ -180,7 +180,11 @@ export class PresetsController {
         preset.setOriginalUserPresetsFile(folder, presetsFile);
     };
 
-    private async resetPresetsFile(file: string, setPresetsFile: SetPresetsFileFunc, setOriginalPresetsFile: SetPresetsFileFunc, fileExistCallback: (fileExists: boolean) => void) {
+    private async resetUserPresetsFile(file: string, setPresetsFile: SetPresetsFileFunc, setOriginalPresetsFile: SetPresetsFileFunc, fileExistCallback: (fileExists: boolean) => void) {
```

Comment:
You don't need the split the function and add a boolean to track this state.  The `setPresetsFile` arg is a function that calls either `_setPresetsFile` or `_setUserPresetsFile`.  You should set the `isUserPreset` fields in that function instead.